UI/window-basic-media-controls.hpp

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <QDialog>
+#include <QPushButton>
+#include <QHBoxLayout>
+#include <QSlider>
+#include <QLabel>
+#include <QTimer>
+#include <obs.hpp>
+
+class OBSBasic;
+
+class OBSBasicMediaControls : public QDialog {
+	Q_OBJECT
+
+private:
+	OBSSceneItem  item;
```

Comment:
You may want to store an `OBSSource`, or better, `OBSWeakSource` instead of `OBSSceneItem`.

`OBSWeakSource` will require conversion to `OBSSource` every time you want to use it, but it will allow you to hold a reference without having actual ownership, and would allow you to very easily detect when the source has been destroyed/removed.  If it's converted and you get a null value for the `OBSSource`, then it's destroyed, and the window should probably terminate.

Alternatively if you really want to hold a reference, you need to at least check for removal, similarly to how the `OBSBasicProperties` and `OBSBasicFilters` dialogs work.  Those hook the source's "remove" signal.  However, I think `OBSWeakSource` is more than adequate for your case and both easier and more ideal overall.